src/lib.rs

```diff
@@ -2683,18 +2683,21 @@ impl<'a, L: 'static + Ledger, Backend: 'a + WalletBackend<'a, L> + Send + Sync>
         let mutex = self.mutex.clone();
         self.task_scope.spawn_cancellable(
             async move {
-                while let Some((event, source)) = events.next().await {
+                let mut finished = false;
+                while !finished {
+                    let next_event = events.next().await.unwrap();
+
                     let WalletSharedState {
                         state,
                         session,
                         pending_key_scans,
                         ..
                     } = &mut *mutex.lock().await;
-                    let finished = if let Some((key, ScanOutputs { records, history })) = state
+                    finished = if let Some((key, ScanOutputs { records, history })) = state
```

Comment:
This whole loop seems like a noop refactor, since we still wait on an event before updating the scan

Reply:
Yeah, at this point it's just a bit of tidying, since I prefer
```rs
while !cond {
}
```
to
```rs
loop {
  if cond {
    return;
  }
}
unreachable!()

Reply:
Yeah, I agree this is a slight improvement. Are you going to fix the issue with always requiring an initial event in a separate PR though? Or change Espresso so that it always creates at least one event? What's the plan?